yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnShuffleEncryptionSuite.scala

```diff
@@ -0,0 +1,226 @@
+        var in: InputStream = new FileInputStream(mergedOutputFile)
+        ByteStreams.skipFully(in, startOffset)
+        in = new LimitedInputStream(in, partitionSize)
+        if (CryptoConf.isShuffleEncryptionEnabled(conf)) {
+          in = CryptoStreamUtils.createCryptoInputStream(in, conf)
+        }
+        if (conf.getBoolean("spark.shuffle.compress", true)) {
+          in = CompressionCodec.createCodec(conf).compressedInputStream(in)
+        }
+        val recordsStream = serializer.newInstance().deserializeStream(in)
+        val records = recordsStream.asKeyValueIterator
+        while (records.hasNext) {
+          val record = records.next()
+          assertEquals(i - 1, hashPartitioner.getPartition(record._1))
+          recordsList.add(record.asInstanceOf[Tuple2[Object, Object]])
+        }
+        recordsStream.close()
+        startOffset += partitionSize
+      }
+    }
+    recordsList
+  }
+
+  private[this] def initialShuffleWriter(): Unit = {
+    val keys = new Array[Byte](16)
+    new Random().nextBytes(keys)
+    val creds = new Credentials()
+    creds.addSecretKey(CryptoConf.SPARK_SHUFFLE_TOKEN, keys)
+    SparkHadoopUtil.get.addCurrentUserCredentials(creds)
+    when(blockManager.diskBlockManager).thenReturn(diskBlockManager)
+    when(blockManager.conf).thenReturn(conf)
+    when(blockManager.getDiskWriter(any(classOf[BlockId]), any(classOf[File]),
+      any(classOf[SerializerInstance]), anyInt, any(classOf[ShuffleWriteMetrics]))).thenAnswer(
+          new Answer[DiskBlockObjectWriter]() {
+            @throws(classOf[Throwable])
+            override def answer(invocationOnMock: InvocationOnMock): DiskBlockObjectWriter = {
+              val args = invocationOnMock.getArguments()
+              new DiskBlockObjectWriter(args(1).asInstanceOf[File],
+                args(2).asInstanceOf[SerializerInstance],
+                args(3).asInstanceOf[Integer], new CompressStream(), false,
+                args(4).asInstanceOf[ShuffleWriteMetrics], args(0).asInstanceOf[BlockId],
+                conf)
+            }
+          })
+
+    when(blockResolver.getDataFile(anyInt(), anyInt())).thenReturn(mergedOutputFile);
+    doAnswer(new Answer[Unit]() {
+      @throws(classOf[Throwable])
+      override def answer(invocationOnMock: InvocationOnMock): Unit = {
+        partitionSizesInMergedFile = (invocationOnMock.getArguments()(2)).asInstanceOf[Array[Long]]
+        val tmp = invocationOnMock.getArguments()(3)
+        mergedOutputFile.delete()
+        tmp.asInstanceOf[File].renameTo(mergedOutputFile)
+      }
+    }).when(blockResolver).writeIndexFileAndCommit(anyInt(), anyInt(), any(classOf[Array[Long]]),
+          any(classOf[File]))
+
+    when(diskBlockManager.createTempShuffleBlock()).thenAnswer(
+      new Answer[Tuple2[TempShuffleBlockId, File]]() {
+        @throws(classOf[Throwable])
+        override def answer(invocationOnMock: InvocationOnMock): Tuple2[TempShuffleBlockId, File]
```

Comment:
Can you say `(TempShuffleBlockId, File)` instead of `Tuple2[TempShuffleBlockId, File]`? Then the `= {` fits in this line.